src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -825,6 +826,16 @@ public final AuthenticationConfiguration useRealm(String realm) {
         return new SetRealmAuthenticationConfiguration(this, realm);
     }
 
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given {@link BearerTokenCredential} to authenticate.
+     *
+     * @param credential the bearer token credential to use
+     * @return the new configuration
+     */
+    public final AuthenticationConfiguration useBearerTokenCredential(BearerTokenCredential credential) {
+        return useCredentials(IdentityCredentials.NONE.withCredential(credential));
```

Comment:
Yeah your example is correct.  I don't want to have too close an overlap between those two types but I'll look at it some more.  For now I'd stick with what you've asked about above.